standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -577,7 +577,7 @@ public RawStore getMS() throws MetaException {
     return getMSForConf(conf);
   }
 
-  public static RawStore getMSForConf(Configuration conf) throws MetaException {
+  public static synchronized RawStore getMSForConf(Configuration conf) throws MetaException {
```

Comment:
can you please clarify how is this synchronization related to the solution?

Reply:
This is another flakyness. Without this, org.apache.hadoop.hive.ql.txn.compactor.TestCompactionMetrics.testInitiatorFailuresCountedCorrectly deadlocked repeatedly on my machine. 4 threads were trying to init the DB (and get the RawStore) concurrently at the smae time which resulted in a deadlock.